closed/adds/jdk/src/windows/native/jdk/crypto/jniprovider/NativeCrypto_md.c

```diff
@@ -61,3 +88,108 @@ void * find_crypto_symbol(void *handle, const char *symname) {
     return symptr;
 }
 
+/*
+ * Copyright (c) 1997, 2015, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.  Oracle designates this
+ * particular file as subject to the "Classpath" exception as provided
+ * by Oracle in the LICENSE file that accompanied this code.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * Find path to JRE based on .exe's location
+ */
+jboolean
+GetJREPath(char *path, jint pathsize)
+{
+    if (GetApplicationHome(path, pathsize)) {
+    
+        char javadll[MAX_PATH];
+        struct stat s;
+        int rc;
+
+        /* Is JRE co-located with the application? */
+        rc = JLI_Snprintf(javadll, sizeof(javadll), "%s\\bin\\" JAVA_DLL, path);
+        if ((rc > 0) && (rc <= MAX_PATH) && (stat(javadll, &s) == 0)) {
```

Comment:
These tests should be `rc < MAX_PATH` to be sure the trailing NUL wasn't written over part of the path.